internal/pkg_web/assembler.js

```diff
@@ -29,9 +30,47 @@ function mkdirp(p) {
   }
 }
 
+function getMappingsFromVolatileFile(volatileFilePath) {
+  const stampFileLines = fs.readFileSync(volatileFilePath, {encoding: 'utf-8'}).trim().split('\n');
+  const stampMap = {};
+  for (const line of stampFileLines) {
+    const [key, value] = line.split(' ');
+    stampMap[key] = value;
+  }
+  return stampMap;
+}
+
+function normalizeSubstitutions(substitutionsArg, volatileFilePath) {
+  const substitutions = JSON.parse(substitutionsArg);
+  const stampMap = getMappingsFromVolatileFile(volatileFilePath);
+
+  const normalizedSubstitutions = {};
+
+  for (const occurrence in substitutions) {
+    let substituteWith = substitutions[occurrence];
+    if (substituteWith.match(/^{.*?}$/)) {
+      substituteWith = substituteWith.replace(/^{(.*?)}$/, '$1');
+      if (!stampMap[substituteWith]) {
+        throw new Error(`Could not find ${substituteWith} key in volatile-status file.`);
```

Comment:
I'm not sure about throwing this error here - if the user is replacing `0.0.0-PLACEHOLDER` as the version string for regular builds with a key from `--workspace_status_command` which is tied to `--stamp`, this will throw when not running a stamped build

Reply:
When I tie this with the version file from node context, which only appears when --stamp option is present, this would be long gone. So I thought it would be better if we explicitly check for the keys when the user trying to stamp builds with --stamp option present.